File: app/src/main/java/forpdateam/ru/forpda/model/repository/MentionsRepository.java
package forpdateam.ru.forpda.model.repository;

import forpdateam.ru.forpda.api.mentions.Mentions;
import forpdateam.ru.forpda.api.mentions.models.MentionsData;
import forpdateam.ru.forpda.model.system.SchedulersProvider;
import io.reactivex.Observable;

/**
 * Created by radiationx on 01.01.18.
 */

public class MentionsRepository {
    private Mentions mentionsApi;
    private SchedulersProvider schedulers;

    public MentionsRepository(SchedulersProvider schedulers, Mentions mentionsApi) {
        this.mentionsApi = mentionsApi;
        this.schedulers = schedulers;
    }

    public Observable<MentionsData> getMentions(int page) {
        return Observable.fromCallable(() -> mentionsApi.getMentions(page))
                .subscribeOn(schedulers.io())
                .observeOn(schedulers.ui());
    }
}

File: app/src/main/java/forpdateam/ru/forpda/model/repository/mentions/MentionsRepository.kt
package forpdateam.ru.forpda.model.repository.mentions

import forpdateam.ru.forpda.entity.remote.mentions.MentionsData
import forpdateam.ru.forpda.model.data.remote.api.mentions.MentionsApi
import ru.radiationx.coretypes.PageOffset
import javax.inject.Inject

/**
 * Created by radiationx on 01.01.18.
 */

class MentionsRepository @Inject constructor(
    private val mentionsApi: MentionsApi
) {

    suspend fun getMentions(offset: PageOffset): MentionsData {
        return mentionsApi.getMentions(offset)
    }

}
